### DISCUSSION TITLE
Global revalidating status available?

### DISCUSSION THREAD

**Discussion by pke** [vercel/swr#1068] [Q&A]
I wonder if there is a way to display a global revalidating status for all pending requests.
I'd like to show a little indicator in the app when transfers happen.

**Comment by ErnestBrandi** [comment]
Hi, this is what you mean ? 
```
const { data1 } = useSWR('key1', fetcher);
const { data2 } = useSWR('key2', fetcher);

if(!data1 && !data2) return <>Loading...</>;
return <>{data1}{data2}</>;
```

**Reply by pke** [reply]
No. On a global level like where I could monitor how many hooks are revalidating at the moment. 

Swr is already monitoring slow connections somehow so I thought it  also knows how many revalidation is happening at any time.

**Reply by ErnestBrandi** [reply]
Then you better check the source code. They do seem to have a CONCURRENT_PROMISES record, but it is not exported. 
That being said I cannot really see a good use case of the monitoring system you think of; it sounds like a blocking functionality to me.

**Comment by kmashint** [comment]
I was looking for something that could be leveraged to detect this by logic of the surrounding application from a function on the SWRConfig.  The SWRConfig and general options allow for onSuccess: function, onError: function, but they don't have an onRevalidate: function to detect when a Promise cycle starts that ends with onSuccess or onError.  If there was an onRevalidate: function option, then application could track the beginning and ending of revalidations and interpret as necessary.  The effect I'm looking for is an overall loading spinner when some non-trivial (more than 0.5 or 1 second) loading/validating is happening.
